core/src/test/java/com/backblaze/b2/json/B2JsonBoundedByteArrayOutputStreamTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+package com.backblaze.b2.json;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+
+import static junit.framework.Assert.assertNull;
+import static junit.framework.TestCase.assertEquals;
+
+/**
+ * Unit test for B2JsonByteArrayOutputStream
+ */
+public class B2JsonBoundedByteArrayOutputStreamTest {
+
+    /**
+     * Cannot directly test the case with MAX_ARRAY_SIZE being (Integer.MAX_VALUE - 8)
```

Comment:
It would be cleaner to pass the maximum into the constructor.  I think having the caller explicitly specify the max is good, then the callers know what's going on.